### PR TITLE
refined spreadtickets calculations

### PR DESCRIPTION
**Pull request by raedah**
-Improve calculation for redeemed funds becoming available by figuring out the users average ticket price in the pool
-Take into account stake rewards given from redeemed tickets. Looks up the stake vote reward for this.

This passes manual testing on testnet

### CODE REVIEW

**Comment by raedah** [comment]
review requested

**Comment by raedah** [comment]
this is ready for review

File: ticketbuyer/buyer.go [hunk 1]
```diff
@@ -454,34 +464,49 @@ func (t *TicketPurchaser) Purchase(height int64) (*PurchaseStats, error) {
...
+		log.Debugf("Your average purchase price for tickets in the pool is %.2f DCR", yourAvgTixPrice)
+		log.Debugf("Available funds of %.2f DCR can buy %.2f tickets, %.2f tickets per block",
+			bal.Spendable.ToCoin()-balanceToMaintainAmt.ToCoin(), tixCanBuy, buyPerBlock)
+		log.Debugf("With %.2f%% proportion live, you will redeem ~%.2f tickets in the remaining %d blocks",
+			proportionLive*100, tixWillRedeem, blocksRemaining)
+		log.Debugf("Redeemed ticket value expected is %.2f DCR, buys %.2f tickets, %.2f%% more",
+			redeemedFunds, tixToBuyWithRedeemedFunds, tixToBuyWithRedeemedFunds/tixCanBuy*100)
+		log.Debugf("Stake reward expected is %.2f DCR, buys %.2f tickets, %.2f%% more",
+			stakeRewardFunds, tixToBuyWithStakeRewardFunds, tixToBuyWithStakeRewardFunds/tixCanBuy*100)
+		log.Infof("Will buy ~%.2f tickets per block, %.2f ticket purchases remain this window", buyPerBlockAll, tixCanBuyAll)
```
Comment (alexlyp) [inline ticketbuyer/buyer.go]: Got "Will buy ~NaN tickets per block, NaN ticket purchases remain this window.

File: ticketbuyer/buyer.go [hunk 2]
```diff
@@ -454,34 +464,49 @@ func (t *TicketPurchaser) Purchase(height int64) (*PurchaseStats, error) {
...
 
+		// same as proportionlive that getstakeinfo rpc shows
+		proportionLive := float64(t.stakeLive) / float64(t.stakePoolSize)
 		// Number of blocks remaining to purchase tickets in this window
 		blocksRemaining := int(winSize) - t.idxDiffPeriod
 		// Estimated number of tickets you will vote on and redeem this window
-		tixWillRedeem := float64(blocksRemaining) * float64(t.activeNet.TicketsPerBlock) * t.ProportionLive
+		tixWillRedeem := float64(blocksRemaining) * float64(t.activeNet.TicketsPerBlock) * proportionLive
+		// Average price of your tickets in the pool
+		yourAvgTixPrice := bal.LockedByTickets.ToCoin() / float64(t.stakeLive+t.stakeImmature)
```
Comment (alexlyp) [inline ticketbuyer/buyer.go]: So the below comment about the Nan's appear to be related to this.  Need a check if t.stakeLive && t.stakeImmature == 0

Comment (alexlyp) [inline ticketbuyer/buyer.go]: Just confirmed that it works again with the check

Comment (raedah) [inline ticketbuyer/buyer.go]: that is added

File: ticketbuyer/buyer.go [hunk 3]
```diff
@@ -454,34 +464,52 @@ func (t *TicketPurchaser) Purchase(height int64) (*PurchaseStats, error) {
...
+		// same as proportionlive that getstakeinfo rpc shows
+		proportionLive := float64(t.stakeLive) / float64(t.stakePoolSize)
 		// Number of blocks remaining to purchase tickets in this window
 		blocksRemaining := int(winSize) - t.idxDiffPeriod
 		// Estimated number of tickets you will vote on and redeem this window
-		tixWillRedeem := float64(blocksRemaining) * float64(t.activeNet.TicketsPerBlock) * t.ProportionLive
+		tixWillRedeem := float64(blocksRemaining) * float64(t.activeNet.TicketsPerBlock) * proportionLive
+		// Average price of your tickets in the pool
+		yourAvgTixPrice := 0.0
+		if t.stakeLive+t.stakeImmature == 0 {
```
Comment (alexlyp) [inline ticketbuyer/buyer.go]: if != 0

Comment (raedah) [inline ticketbuyer/buyer.go]: doh!

**Pull request merged**